test/server/server_test.cc

```diff
@@ -642,6 +642,48 @@ TEST_P(ServerInstanceImplTest, DEPRECATED_FEATURE_TEST(LoadsV2BootstrapFromPbTex
   EXPECT_FALSE(server_->localInfo().node().hidden_envoy_deprecated_build_version().empty());
 }
 
+// Validate that bootstrap v3 pb_text with new fields loads fails if V2 config is specified.
+TEST_P(ServerInstanceImplTest, FailToLoadV3ConfigWhenV2SelectedFromPbText) {
+  options_.bootstrap_version_ = 2;
+
+  EXPECT_THROW_WITH_REGEX(
+      initialize("test/server/test_data/server/valid_v3_but_invalid_v2_bootstrap.pb_text"),
+      EnvoyException, "Unable to parse file");
+}
+
+// Validate that we correctly parse a V2 file when configured to do so.
+TEST_P(ServerInstanceImplTest, LoadsV2ConfigWhenV2SelectedFromPbText) {
```

Comment:
Hmm this seems to fail under `compile_time_options` with

```
[ RUN      ] IpVersions/ServerInstanceImplTest.LoadsV2ConfigWhenV2SelectedFromPbText/IPv4
[libprotobuf WARNING external/com_google_protobuf/src/google/protobuf/text_format.cc:324] Warning parsing text-format envoy.config.bootstrap.v2.Bootstrap: 4:1: text format contains deprecated field "build_version"
[2020-04-22 05:18:57.856][17][critical][main] [external/envoy/source/server/server.cc:98] error initializing configuration '/b/f/w/_tmp/fa9accb9e25432420d3f5adcb6c78680/valid_v2_but_invalid_v3_bootstrap.pb_text.with.ports.pb_text': Proto constraint validation failed (Using deprecated option 'envoy.api.v2.core.Node.build_version' from file base.proto. This configuration will be removed from Envoy soon. Please see https://www.envoyproxy.io/docs/envoy/latest/intro/deprecated for details. If continued use of this field is absolutely necessary, see https://www.envoyproxy.io/docs/envoy/latest/configuration/operations/runtime#using-runtime-overrides-for-deprecated-features for how to apply a temporary and highly discouraged override.): id: "bootstrap_id"
build_version: "foo"

unknown file: Failure
C++ exception with description "Proto constraint validation failed (Using deprecated option 'envoy.api.v2.core.Node.build_version' from file base.proto. This configuration will be removed from Envoy soon. Please see https://www.envoyproxy.io/docs/envoy/latest/intro/deprecated for details. If continued use of this field is absolutely necessary, see https://www.envoyproxy.io/docs/envoy/latest/configuration/operations/runtime#using-runtime-overrides-for-deprecated-features for how to apply a temporary and highly discouraged override.): id: "bootstrap_id"
build_version: "foo"
" thrown in the test body.
[  FAILED  ] IpVersions/ServerInstanceImplTest.LoadsV2ConfigWhenV2SelectedFromPbText/IPv4, where GetParam() = 4-byte object <00-00 00-00> (4 ms)
```

presumably since we run the test with `--disabled-features=disabled`. Thoughts on how to make this test work @htuch ?

Reply:
Can this be a `DEPRECATED_FEATURE_TEST`?

Reply:
Yep this worked, thanks!